### PR TITLE
Feature: Cli download commands aleph & ipfs

### PR DESCRIPTION
**Pull request by 1yam**
Solutions:
   - Create a new command that allows users to download a file from a hash.

### CODE REVIEW

File: src/aleph_client/commands/files.py [hunk 1]
```diff
@@ -98,3 +98,34 @@ def upload(
...
+    if path is None:
+        path = item_hash
+
+    with AlephClient(api_server=sdk_settings.API_HOST) as client:
+        try:
+            typer.echo("Downloading file ...")
+            if not use_ipfs:
+                file_content = client.download_file(item_hash)
+            else:
+                file_content = client.download_file_ipfs(item_hash)
```
Comment (odesenfans) [inline src/aleph_client/commands/files.py:126]: You can now use `download_file_to_buffer` to reduce memory usage.

File: src/aleph_client/commands/utils.py [hunk 2]
```diff
@@ -122,3 +123,14 @@ def str_to_datetime(date: Optional[str]) -> Optional[datetime]:
...
+
+def write_file_from_bytes(
+    path: str,
+    content: bytes,
+):
+    """
+    write content of file download to byte into a path
+    """
+    file_path = Path(path)
+    file_path.write_bytes(content)
```
Comment (odesenfans) [inline src/aleph_client/commands/utils.py:136]: This function will be useless once you use `download_file_to_buffer`.

**Review by odesenfans — CHANGES_REQUESTED** [review]

File: src/aleph_client/commands/utils.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 from pygments.lexers import JsonLexer
 from typer import echo
 from datetime import datetime
+from pathlib import Path
```
Comment (MHHukiewitz) [inline src/aleph_client/commands/utils.py:10]: unneeded import

File: src/aleph_client/commands/utils.py [hunk 4]
```diff
@@ -122,3 +123,4 @@ def str_to_datetime(date: Optional[str]) -> Optional[datetime]:
     except ValueError:
         pass
     return datetime.fromisoformat(date)
+
```
Comment (MHHukiewitz) [inline src/aleph_client/commands/utils.py:126]: you can revert this file

File: tests/unit/test_utils.py [hunk 5]
```diff
@@ -15,4 +15,4 @@ def test_get_message_type_value():
     assert get_message_type_value(AggregateMessage) == MessageType.aggregate
     assert get_message_type_value(StoreMessage) == MessageType.store
     assert get_message_type_value(ProgramMessage) == MessageType.program
-    assert get_message_type_value(ForgetMessage) == MessageType.forget
+    assert get_message_type_value(ForgetMessage) == MessageType.forget
```
Comment (MHHukiewitz) [inline tests/unit/test_utils.py:18]: seems also unneeded? Maybe it's due to auto formatting

**Review by MHHukiewitz — CHANGES_REQUESTED** [review]
LGTM, only minor formating issues, use isort, flake8 & black

File: src/aleph_client/commands/files.py [hunk 6]
```diff
@@ -99,3 +102,36 @@ async def upload(
...
+):
+    """Download a file on aleph.im."""
+
+    setup_logging(debug)
+
+    # If no path given then path == item_hash
+    if path is None:
+        path = item_hash
+
+    with AlephClient(api_server=sdk_settings.API_HOST) as client:
```
Comment (MHHukiewitz) [inline src/aleph_client/commands/files.py:124]: This is the wrong import, try using `async with AlephHttpClient`

**Comment by MHHukiewitz** [comment]
Closing this PR and will be superseded by a fresh branch. GH seems to not handle some of the rebases/force-pushes well that happened here

**Pull request closed without merging**